### DISCUSSION TITLE
fromSnapshot equivalent to new?

### DISCUSSION THREAD

**Discussion by maccman** [xaviergonz/mobx-keystone#229] [General]
Question: I've instantiating a model from some attributes fetched from the server. I've switched to using a constructor rather than `fromSnapshot` because I want the defaults props to be set.

Is `new Note(serverProps)` practically the same as `fromSnapshot<Note>(serverProps)`?

**Comment by xaviergonz** [comment]
Yes, except from snapshot requires $modelType / $modelId and submodels are plain objects for fromSnapshot while they have to be actual instances for the constructor.

**Comment by xaviergonz** [comment]
Btw, the default props should be set either way
